docs/data/toolpad/core/components/crud/CrudNonEditableFields.js

```diff
@@ -0,0 +1,285 @@
+    { field: 'title', headerName: 'Title', flex: 1 },
+    { field: 'text', headerName: 'Text', flex: 1 },
+    {
+      field: 'createdAt',
+      headerName: 'Created at',
+      type: 'dateTime',
+      valueGetter: (value) => value && new Date(value),
+      editable: false,
+    },
+  ],
+  // preview-end
+  getMany: async ({ paginationModel, filterModel, sortModel }) => {
+    // Simulate loading delay
+    await new Promise((resolve) => {
+      setTimeout(resolve, 750);
+    });
+
+    let processedNotes = [...notesStore];
+
+    // Apply filters (demo only)
+    if (filterModel?.items?.length) {
+      filterModel.items.forEach(({ field, value, operator }) => {
+        if (!field || value == null) {
+          return;
+        }
+
+        processedNotes = processedNotes.filter((note) => {
+          const noteValue = note[field];
+
+          switch (operator) {
+            case 'contains':
+              return String(noteValue)
+                .toLowerCase()
+                .includes(String(value).toLowerCase());
+            case 'equals':
+              return noteValue === value;
+            case 'startsWith':
+              return String(noteValue)
+                .toLowerCase()
+                .startsWith(String(value).toLowerCase());
+            case 'endsWith':
+              return String(noteValue)
+                .toLowerCase()
+                .endsWith(String(value).toLowerCase());
+            case '>':
+              return noteValue > value;
+            case '<':
+              return noteValue < value;
+            default:
+              return true;
+          }
+        });
+      });
+    }
+
+    // Apply sorting
+    if (sortModel?.length) {
+      processedNotes.sort((a, b) => {
+        for (const { field, sort } of sortModel) {
+          if (a[field] < b[field]) {
+            return sort === 'asc' ? -1 : 1;
+          }
+          if (a[field] > b[field]) {
+            return sort === 'asc' ? 1 : -1;
+          }
+        }
+        return 0;
+      });
+    }
+
+    // Apply pagination
+    const start = paginationModel.page * paginationModel.pageSize;
+    const end = start + paginationModel.pageSize;
+    const paginatedNotes = processedNotes.slice(start, end);
+
+    return {
+      items: paginatedNotes,
+      itemCount: processedNotes.length,
+    };
+  },
+  getOne: async (noteId) => {
+    // Simulate loading delay
+    await new Promise((resolve) => {
+      setTimeout(resolve, 750);
+    });
+
+    const noteToShow = notesStore.find((note) => note.id === Number(noteId));
+
+    if (!noteToShow) {
+      throw new Error('Note not found');
+    }
+    return noteToShow;
+  },
+  createOne: async (data) => {
+    // Simulate loading delay
+    await new Promise((resolve) => {
+      setTimeout(resolve, 750);
+    });
+
+    const newNote = {
+      id: notesStore.reduce((max, note) => Math.max(max, note.id), 0) + 1,
+      createdAt: new Date().toISOString(),
```

Comment:
In the docs example this `createdAt` doesn't seem to be working... Will fix separately now that I already merged it.